algo/src/main/java/org/neo4j/graphalgo/MSTPrimProc.java

```diff
@@ -43,36 +46,48 @@
 
         ProcedureConfiguration configuration = ProcedureConfiguration.create(config);
 
-        LazyIdMapper idMapper = new LazyIdMapper();
+        LazyIdMapper idMapper = LazyIdMapper.importer(api)
+                .withWeightsFromProperty(propertyName, 1.0)
+                .withOptionalLabel(configuration.getNodeLabelOrQuery())
+                .withOptionalRelationshipType(configuration.getRelationshipOrQuery())
+                .build();
 
         MSTPrimResult.Builder builder = MSTPrimResult.builder();
 
-        ProgressTimer timer = ProgressTimer.start(builder::withLoadDuration);
-        BufferedWeightMap weightMap = BufferedWeightMap.importer(api)
+        CompletableFuture<BufferedWeightMap> weightMap = BufferedWeightMap.importer(api)
```

Comment:
I think we should move the code out of the procedure into the algorithm or into a dedicated (reusable?) parallel loader?